src/search/__init__.py

```diff
@@ -0,0 +1,354 @@
+        """Search async execute.
+
+        Parameters:
+            context: Blender context
+
+        Returns:
+            enum set in {‘RUNNING_MODAL’, ‘CANCELLED’, ‘FINISHED’}
+        """
+        logging.debug('STARTING ASYNC SEARCH')
+        search_props = search.get_search_props()
+        if self.author_id != '':
+            search_props.search_keywords = ''
+        if self.keywords != '':
+            search_props.search_keywords = self.keywords
+
+        logging.debug(f'GOT SEARCH_PROPS: {search_props}')
+        ui = UI()
+        ui_props = getattr(bpy.context.window_manager, HANA3D_UI)
+        asset_type = ui_props.asset_type.lower()
+
+        logging.debug('CREATING QUERY OBJECT')
+        query = Query(bpy.context, search_props)
+        query.asset_type = asset_type
+        logging.debug(f'GOT QUERY OBJECT: {query.to_dict()}')
+
+        if search_props.is_searching and self.get_next:
+            return {'FINISHED'}
+
+        search_props.is_searching = True
+
+        options = {'get_next': self.get_next}
+        ui.add_report(text=f'{HANA3D_DESCRIPTION} searching...', timeout=2)
+
+        request_data = await search_assets(query, options, ui)
+
+        tempdir = paths.get_temp_dir(f'{asset_type}_search')
+        logging.debug(f'GOT TEMPDIR: {tempdir}')
+
+        result_field = []
+        ok, error = self._check_errors(request_data)
+        logging.debug(f'GOT TEMPDIR: {tempdir}')
+        if ok:
+            run_assetbar_op = getattr(bpy.ops.object, f'{HANA3D_NAME}_run_assetbar_fix_context')
+            status = run_assetbar_op()
+            logging.debug(f'RAN ASSET_BAR: {status}')
+
+            result_field = self._parse_response(asset_type, request_data)
+            logging.debug(f'PARSED RESULTS: {len(result_field)}')
+
+            search.set_search_results(asset_type, result_field)
+            search.set_original_search_results(asset_type, request_data)
+            logging.debug('SET SEARCH_RESULTS')
+
+            search.load_previews(asset_type, result_field)
+            logging.debug('LOADED PREVIEWS')
+
+            if len(result_field) < ui_props.scrolloffset:
+                ui_props.scrolloffset = 0
+            search_props.is_searching = False
+            text = f'Found {request_data["count"]} results. '  # noqa #501
+            ui.add_report(text=text)
+        else:
+            logging.error(error)
+            ui = UI()
+            ui.add_report(text=error, color=colors.RED)
+            seach_props.search_error = True
+
+        # we save here because a missing thumbnail check is in the previous loop
+        # we can also prepend previous results. These have downloaded thumbnails already...
+        if options['get_next']:
+            request_data['results'][0] = request_data['results']
+
+        json_filepath = os.path.join(tempdir, f'{asset_type}_searchresult.json')
+        logging.debug(f'SAVING RESULTS ON DISK: {json_filepath}')
```

Comment:
acho que isso aqui não é mais necessário 🤔 
se eu entendi bem, isso era usado p/ passar informação entre threads
agora que tá tudo via asyncio acho que poderia ser feito de outra maneira menos gambiarra

Reply:
Também acho que não, vou checar se nenhum outro processo precisa usar e se for o caso já tiro